### ISSUE TITLE
Inference Issue

### ISSUE THREAD

**Issue by gunjanmimo** [wysoczanska/clip_dinoiser#10]
While trying the demo.ipynb and demo.py with the given example image and checkpoint file, I got this error 

```
RuntimeError: Given groups=1, weight of size [512, 768, 1, 1], expected input[1, 0, 24, 36] to have 768 channels, but got 0 channels instead
```

**Comment by wysoczanska** [comment]
Hi! Thanks for the info. Could you please provide the whole error trace? Also if you could give me details of your environment that would be great. Thanks!

**Comment by gunjanmimo** [comment]
Here is the error I am getting when I am trying to run the demo.ipynb

```
{
	"name": "RuntimeError",
	"message": "Given groups=1, weight of size [512, 768, 1, 1], expected input[1, 0, 24, 36] to have 768 channels, but got 0 channels instead",
	"stack": "---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[4], line 13
     10 model.apply_found = True
     12 # run segmentation
---> 13 mask, ticks, img = visualize_per_image(file, TEXT_PROMPTS, PALETTE, model)
     15 fig, ax = plt.subplots(nrows=1, ncols=2)
     16 alpha=0.5

Cell In[2], line 8, in visualize_per_image(file_path, text_prompts, palette, model)
      5 img_tens = T.PILToTensor()(img).unsqueeze(0).to(device) / 255.
      7 h, w = img_tens.shape[-2:]
----> 8 output = model(img_tens).cpu()
      9 output = F.interpolate(output, scale_factor=model.vit_patch_size, mode=\"bilinear\",
     10                        align_corners=False)[..., :h, :w]
     11 output = output[0].argmax(dim=0)

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~/workspace/clip_dinoiser/models/clip_dinoiser/clip_dinoiser.py:274, in CLIP_DINOiser.forward(self, x)
    273 def forward(self, x: torch.Tensor):
--> 274     preds, corrs, output = self.forward_pass(x)
    275     B, C, hf, wf = output.shape
    276     preds = F.interpolate(preds, (hf, wf), mode=\"bilinear\", align_corners=False)

File ~/workspace/clip_dinoiser/models/clip_dinoiser/clip_dinoiser.py:257, in CLIP_DINOiser.forward_pass(self, x)
    255 def forward_pass(self, x: torch.Tensor):
    256     x = self.make_input_divisible(x)
--> 257     clip_proj_feats = self.get_clip_features(x)[0]
    258     B, c_dim, h, w = clip_proj_feats.shape
    259     if self.feats_idx != 'final':

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/utils/_contextlib.py:115, in context_decorator.<locals>.decorate_context(*args, **kwargs)
    112 @functools.wraps(func)
    113 def decorate_context(*args, **kwargs):
    114     with ctx_factory():
--> 115         return func(*args, **kwargs)

File ~/workspace/clip_dinoiser/models/clip_dinoiser/clip_dinoiser.py:172, in DinoCLIP.get_clip_features(self, x)
    166 \"\"\"
    167 Extracts MaskCLIP features
    168 :param x: (torch.Tensor) - batch of input images
    169 :return: (torch.Tensor) - clip dense features, (torch.Tensor) - output probabilities
    170 \"\"\"
    171 x = self.make_input_divisible(x)
--> 172 maskclip_map, feat = self.clip_backbone(x, return_feat=True)
    174 return feat, maskclip_map

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~/workspace/clip_dinoiser/models/maskclip/maskclip.py:126, in MaskClip.forward(self, inputs, return_feat)
    124 x = self.extract_feat(inputs)
    125 if return_feat:
--> 126     seg_logits, feats = self.decode_head(x, return_feat)
    127     return seg_logits, feats
    128 else:

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~/workspace/clip_dinoiser/models/maskclip/maskclip.py:181, in MaskClipHead.forward(self, inputs, return_feat)
    179 def forward(self, inputs, return_feat=False):
    180     v = inputs
--> 181     feat = self.proj(v)
    182     output = self.cls_seg(feat)
    183     if return_feat:

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/conv.py:463, in Conv2d.forward(self, input)
    462 def forward(self, input: Tensor) -> Tensor:
--> 463     return self._conv_forward(input, self.weight, self.bias)

File /opt/conda/envs/dino/lib/python3.10/site-packages/torch/nn/modules/conv.py:459, in Conv2d._conv_forward(self, input, weight, bias)
    455 if self.padding_mode != 'zeros':
    456     return F.conv2d(F.pad(input, self._reversed_padding_repeated_twice, mode=self.padding_mode),
    457                     weight, bias, self.stride,
    458                     _pair(0), self.dilation, self.groups)
--> 459 return F.conv2d(input, weight, bias, self.stride,
    460                 self.padding, self.dilation, self.groups)

RuntimeError: Given groups=1, weight of size [512, 768, 1, 1], expected input[1, 0, 24, 36] to have 768 channels, but got 0 channels instead"
}
```

**Comment by guyguy404** [comment]
I met a similar problem, and I solved it by using the following code:
In maskclip.py:
v = v[:, 1:]  -------->  v = v.permute(1, 0, 2)[:, 1:]
In clip_dinoiser.py:
clip_feats = self.train_feats['clip_inter']  -------->  clip_feats = self.train_feats['clip_inter'].permute(1, 0, 2)

**Comment by gunjanmimo** [comment]
Thank you so much @guyguy404. It solved the issue

**Comment by wysoczanska** [comment]
Hi guys! Sorry for late reply, I was taking some time off. Thanks for your message and especially @guyguy404 thanks for your quick solution.

I found a problem, which seems to be a new open-clip-torch version (2.26) which I believe is the one you have in your environments (https://pypi.org/project/open-clip-torch/#history released 2 weeks ago). It looks like there was an update in timm library (I tested the code on 0.9.8 and you most likely have 1.0.7) and the output of Hugging Face (timm) CLIP model has changed, losing backwards compatibility. 

For now, since I won't be able to test all the code with the new open-clip version in the next couple of days, I will freeze the requirements of open-clip version to the one I tested on - to make sure others can run it. 

However, if you were able to reproduce the results and CLIP-DINOiser outputs are unchanged, feel free to create a pull request with suggested changes.

Once again, thanks for the info and sorry for the inconvenience!
